plugins/woocommerce-blocks/assets/js/blocks/product-collection/deprecated.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import { BlockInstance, createBlock } from '@wordpress/blocks';
+import { useInnerBlocksProps, useBlockProps } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { LayoutOptions, ProductCollectionLayout } from './types';
+
+interface DisplayLayoutAttribute {
+	type: 'list' | 'flex';
+	columns: number;
+	shrinkColumns: boolean;
+}
+
+const migrateDisplayLayout = (
+	attributes: { displayLayout: DisplayLayoutAttribute },
+	innerBlocks: BlockInstance[]
+) => {
+	const { displayLayout = null, ...newAttributes } = attributes;
+	if ( ! displayLayout ) {
+		return [ attributes, innerBlocks ];
+	}
+
+	const { type, columns, shrinkColumns } = displayLayout;
+
+	// Convert custom displayLatout to templateLayout which will then
+	// be applied automatically to the template block.
+	let templateLayout: ProductCollectionLayout;
+	if ( type === 'flex' ) {
+		// Convert responsive columns to size-based automatic columns.
+		if ( shrinkColumns ) {
+			// We can replicate the previous responsive behavior with percentage
+			// based minimum width. One consideration, however, is that the
+			// block gap is not accounted for. We can approximate it by
+			// not using 90% as the width for our column area instead.
+			const columnWidth = Math.floor( 90 / columns );
+			templateLayout = {
+				type: LayoutOptions.GRID,
+				minimumColumnWidth: columnWidth + '%',
+			};
```

Comment:
Awesome! I really love how you're using the percentage to keep the existing block compatible. 🚀

Reply:
After being pressed by Karol I made this much more effective at providing correct column breakpoints too!